code/go/0chain.net/smartcontract/storagesc/handler.go

```diff
@@ -880,3 +880,26 @@ func (ssc *StorageSmartContract) GetTotalData(_ context.Context, balances cstate
 
 	return 0, fmt.Errorf("storageSmartContract is nil")
 }
+
+func (ssc *StorageSmartContract) GetCollectedRewards(ctx context.Context, params url.Values, balances cstate.StateContextI) (resp interface{}, err error) {
+	var (
+		startBlock, _ = strconv.Atoi(params.Get("start_block"))
+		endBlock, _   = strconv.Atoi(params.Get("end_block"))
+	)
+
+	query := event.RewardQuery{
+		StartBlock: startBlock,
+		EndBlock:   endBlock,
+	}
```

Comment:
This query will return all the rewards of given blocks range, we don't need to return rewards by client IDs?

Reply:
yes,  we can return by client_id, by range of blocks, or both. I have kept options for other queries as well. Just let me know which one we need.
FYI, I cant test codes from this pr as it is blocked by https://github.com/0chain/0chain/issues/1169